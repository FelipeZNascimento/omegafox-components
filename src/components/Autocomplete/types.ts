export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
