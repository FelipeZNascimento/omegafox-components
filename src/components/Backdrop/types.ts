export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
