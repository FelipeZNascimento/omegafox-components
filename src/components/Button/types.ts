export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
