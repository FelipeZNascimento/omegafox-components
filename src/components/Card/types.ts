export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
