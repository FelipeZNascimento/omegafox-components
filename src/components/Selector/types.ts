export type TSelectorItem = {
  id: number;
  value: number;
  text: string;
};

export interface ISelectorProps {
  items: TSelectorItem[];
  selectedItem?: number | null;
  size?: 'small' | 'big';
  onClick: (itemId: number) => void;
}
