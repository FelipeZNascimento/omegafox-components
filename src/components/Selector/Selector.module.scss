@use "../../styling/colors.scss" as colors;
@use "../../styling/colors_fifa.scss" as colors_fifa;
@use "../../styling/color_utils.scss" as color_utils;

@import "../../styling/alignments.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.container {
  @include flex-aligned(center, center);
  @include padded(bottom, m);

  flex-wrap: wrap;
}

.selectorItem {
  @include aligned(center);
  @include flex-aligned(center, center);
  @include marged(all, s2);
  @include padded(horizontal, s2);

  background-color: colors.$grey1;
  border-radius: $border-radius;
  color: colors.$grey4;
  cursor: pointer;
  font-weight: bold;
  transition: background-color $transition-time, color $transition-time, box-shadow $transition-time;

  &Small {
    height: $selector-height;
    width: $selector-width;
  }

  &Big {
    height: calc($selector-height + 10px);
    width: calc($selector-width + 20px);
  }

  &:hover {
    -webkit-box-shadow: inset 0px 0px 0px 2px colors.$grey3;
    box-shadow: inset 0px 0px 0px 2px colors.$grey3;
    background-color: rgba(colors.$grey1, 0.6);
  }

  &Selected {
    -webkit-box-shadow: inset 0px 0px 0px 2px colors_fifa.$qatar-red;
    box-shadow: inset 0px 0px 0px 2px colors_fifa.$qatar-red;
    background-color: rgba(colors_fifa.$qatar-red, 0.6);
    color: colors.$grey1;
  }

  &:nth-child(4n + 1) {
    border-bottom-left-radius: 0;
  }
  &:nth-child(4n + 2) {
    border-bottom-right-radius: 0;
  }
  &:nth-child(4n + 3) {
    border-top-right-radius: 0;
  }
  &:nth-child(4n + 4) {
    border-top-left-radius: 0;
  }
}
