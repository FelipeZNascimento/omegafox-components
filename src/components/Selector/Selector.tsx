import React from 'react';

import classNames from 'classnames';
import { ISelectorProps } from './types';
import styles from './Selector.module.scss';

export const Selector = ({
  items,
  selectedItem = 0,
  size = 'small',
  onClick
}: ISelectorProps) => {
  return (
    <div className={styles.container}>
      {items.map((item) => {
        const itemClass = classNames(styles.selectorItem, {
          [styles.selectorItemSelected]: selectedItem === item.value,
          [styles.selectorItemSmall]: size === 'small',
          [styles.selectorItemBig]: size === 'big'
        });

        return (
          <div
            key={item.id}
            className={itemClass}
            onClick={() => onClick(item.value)}
          >
            {item.text}
          </div>
        );
      })}
    </div>
  );
};
