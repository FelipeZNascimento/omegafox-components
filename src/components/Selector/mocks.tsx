export const selectorItems = [
  {
    id: 0,
    value: 1,
    text: 'Rodada 1'
  },
  {
    id: 1,
    value: 2,
    text: 'Rodada 2'
  },
  {
    id: 2,
    value: 3,
    text: 'Rodada 3'
  },
  {
    id: 3,
    value: 4,
    text: '2a Fase'
  }
];
