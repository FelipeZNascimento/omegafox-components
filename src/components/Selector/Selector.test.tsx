import React from 'react';
import { render } from '@testing-library/react';
import { Selector } from './Selector';
import { ISelectorProps, TSelectorItem } from './types';

describe('<Selector />', () => {
  const renderComponent = ({ items, selectedItem }: ISelectorProps) =>
    render(
      <Selector items={items} selectedItem={selectedItem} onClick={jest.fn} />
    );

  it('should render', () => {
    const items: TSelectorItem[] = [{ id: 0, value: 1, text: 'text' }];
    const { container } = renderComponent({
      items: items,
      onClick: jest.fn
    });

    expect(container).toMatchSnapshot();
  });
});
