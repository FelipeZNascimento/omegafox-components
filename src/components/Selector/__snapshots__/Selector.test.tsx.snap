// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Selector /> should render 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="selectorItem selectorItemSmall"
    >
      text
    </div>
  </div>
</div>
`;
